aws/resource_aws_cognito_user_pool.go

```diff
@@ -1154,6 +1142,22 @@ func resourceAwsCognitoUserPoolUpdate(d *schema.ResourceData, meta interface{})
 		}
 	}
 
+	if d.HasChange("schema") {
+		oldSchema, newSchema := d.GetChange("schema")
+		if oldSchema.(*schema.Set).Difference(newSchema.(*schema.Set)).Len() == 0 {
+			params := &cognitoidentityprovider.AddCustomAttributesInput{
+				UserPoolId:       aws.String(d.Id()),
+				CustomAttributes: expandCognitoUserPoolSchema(newSchema.(*schema.Set).Difference(oldSchema.(*schema.Set)).List()),
+			}
+			_, err := conn.AddCustomAttributes(params)
+			if err != nil {
+				return fmt.Errorf("unable to add custom attributes from schema: %s", err.Error())
```

Comment:
Nit: to use the %w format for error and some additional context could be helpful to follow pattern above
```
return fmt.Errorf("error updating Cognito User Pool (%s): unable to add custom attributes from schema: %w", d.Id(), err)
```

Reply:
Thanks! That is much more informative!